data_structures/binary_tree/inorder_tree_traversal_2022.py

```diff
@@ -0,0 +1,91 @@
+"""
+Illustrate how to implement inorder traversal in binary search tree.
+Author: Gurneet Singh
+https://www.geeksforgeeks.org/tree-traversals-inorder-preorder-and-postorder/
+"""
+
+
+class BinaryTreeNode:
+    """Defining the structure of BinaryTreeNode"""
+
+    def __init__(self, data: int) -> None:
+        self.data = data
+        self.left_child = None
+        self.right_child = None
+
+
+def insert(node: None | BinaryTreeNode, new_value: int) -> BinaryTreeNode | None:
+    """
+    If the binary search tree is empty, make a new node and declare it as root.
+    >>> node_a = BinaryTreeNode(12345)
+    >>> node_b = insert(node_a, 67890)
+    >>> node_a.left_child == node_b.left_child
+    True
+    >>> node_a.right_child == node_b.right_child
+    True
+    >>> node_a.data == node_b.data
+    True
+    """
+    if node is None:
+        node = BinaryTreeNode(new_value)
+        return node
+
+    # binary search tree is not empty,
+    # so we will insert it into the tree
+    # if new_value is less than value of data in node,
+    #  add it to left subtree and proceed recursively
+    if new_value < node.data:
+        node.left_child = insert(node.left_child, new_value)
+    else:
+        # if new_value is greater than value of data in node,
+        #  add it to right subtree and proceed recursively
+        node.right_child = insert(node.right_child, new_value)
+    return node
+
+
+def inorder(node: None | BinaryTreeNode) -> None:  # if node is None,return
+    """
+    >>> inorder(make_tree())
+    6
+    10
+    14
+    15
+    20
+    25
+    60
+    """
+
+    if node is None:
+        return None
+    # traverse left subtree
+    inorder(node.left_child)
+    # traverse current node
+    print(node.data)
```

Comment:
According the `CONTRIBUTING.md`, algorithmic functions should not `print()`.  See our other tree algorithms for ways to avoid printing in the function.